Правка метода update

pullveryzator opened the pull request:
Changes to be committed:
	modified:   app/crud/base.py

pullveryzator committed:
Правка метода update

Changes to be committed:
	modified:   app/crud/base.py

pullveryzator committed:
Правка импорта.
Changes to be committed:
	modified:   app/crud/base.py

teamofroman commented on src/app/crud/base.py:
```diff
@@ -81,12 +80,11 @@ async def update(
             session: AsyncSession) -> ModelType:
         """Update object in database."""
-        obj_data = jsonable_encoder(db_obj)
         update_data = pydantic_scheme_obj.dict(
```
dict - deprecated метод, вместо него model_dump

pullveryzator replied:
Роман, метод model_dump используется в pydantic 2 и выше, а у нас 1.х Или я ошибаюсь?

teamofroman reviewed (changes requested):
Надо немного поправить

pullveryzator committed:
Добавил новый метод в юзера и поправил create и  update.
On branch fix/crud
Your branch is up to date with 'origin/fix/crud'.

Changes to be committed:
	modified:   app/crud/user_tg.py

Untracked files:
	alembic/versions/2024_11_29_1555-01c576f4b80a_remove_id_field_from_user_group_table.py
	app/templates/

pullveryzator committed:
Добавление получения данных групп для эндпоинтов.
On branch fix/crud
Your branch is up to date with 'origin/fix/crud'.

Changes to be committed:
	modified:   app/crud/user_tg.py

Untracked files:
	alembic/versions/2024_11_29_1555-01c576f4b80a_remove_id_field_from_user_group_table.py
	app/admin_endpoints/
	app/templates/

teamofroman commented on src/app/crud/user_tg.py:
```diff
@@ -22,14 +21,25 @@ async def create(
         await session.commit()
         await session.refresh(new_user)
-        return new_user
+        return {'user': new_user, 'group_names': group_names}
```
Здесь мы создаем пользователя и возвращаем только его. Остальное - вспомогательная информация. Ее возвращать не надо.

teamofroman commented on src/app/crud/user_tg.py:
```diff
@@ -22,14 +21,25 @@ async def create(
-        new_user_dict['hashed_password'] = hash(password)
-        new_user = self.model(**new_user_dict)
+        groups = new_user_dict.pop('groups')
+        new_user_dict['hashed_password'] = str(hash(password))  # TODO:
```
В файле `/src/app/core/jwt.py` есть функции для работы с паролями. Используй их. (Обнови у себя ветку `develop` и подтяни код в свою ветку.

teamofroman commented on src/app/crud/user_tg.py:
```diff
@@ -38,23 +48,33 @@ async def update(
+            Group.id.in_(pydantic_scheme_user.groups)))
+        group_objects = result.scalars().all()
+        group_names = [group.name for group in group_objects]
+        return {'user': db_user, 'group_names': group_names}
```
Возвращаем только пользователя.

teamofroman commented on src/app/crud/user_tg.py:
```diff
@@ -38,23 +48,33 @@ async def update(
-            if hasattr(user_data, field):
-                setattr(db_user, field, update_data[field])
-        session.add(db_user)
+            update_data['hashed_password'] = str(hash(password))  # TODO:
```
В файле /src/app/core/jwt.py есть функции для работы с паролями. Используй их. (Обнови у себя ветку develop и подтяни код в свою ветку.

teamofroman reviewed (changes requested):
Надо немного поправить.

pullveryzator committed:
Merge branch 'develop' of github.com:Studio-Yandex-Practicum/providenie into fix/crud

pullveryzator committed:
Правка возврата create update.

On branch fix/crud
Your branch is up to date with 'origin/fix/crud'.

Changes to be committed:
	modified:   app/crud/user_tg.py

Untracked files:
	alembic/versions/2024_11_29_1555-01c576f4b80a_remove_id_field_from_user_group_table.py
	app/admin_endpoints/
	app/templates/

pullveryzator committed:
Fix get_user_bby_params method
Please enter the commit message for your changes. Lines starting
with '#' will be ignored, and an empty message aborts the commit.

On branch fix/crud
Your branch is up to date with 'origin/fix/crud'.

Changes to be committed:
	modified:   app/crud/user_tg.py

Untracked files:
	alembic/versions/2024_11_29_1555-01c576f4b80a_remove_id_field_from_user_group_table.py
	app/admin_endpoints/
	app/templates/

pullveryzator committed:
Fix user update method

On branch fix/crud
Your branch is ahead of 'origin/fix/crud' by 1 commit.
  (use "git push" to publish your local commits)

Changes to be committed:
	modified:   app/crud/user_tg.py

Untracked files:
	alembic/versions/2024_11_29_1555-01c576f4b80a_remove_id_field_from_user_group_table.py
	app/admin_endpoints/
	app/templates/

teamofroman commented on src/app/crud/user_tg.py:
```diff
@@ -22,11 +22,20 @@ async def create(
+
+        for group in group_objects:
+            association = UserGroupAssociation(
+                user_id=new_user.id, group_id=group.id)
```
`new_user.id` к этому моменту еще не существует. Ты только добавил объект в очередь на создание (session.add(new_user)). Реально в БД его еще нет. Он там появится после коммита. И надо не забыть обновить объект `new_user` из БД (как раз пропишется id созданного пользователя)

pullveryzator replied:
Поправил.

teamofroman reviewed (changes requested):
Совсем чуть-чуть надо поправить

pullveryzator committed:
Fix create method

On branch fix/crud
Your branch is ahead of 'origin/fix/crud' by 2 commits.
  (use "git push" to publish your local commits)

Changes to be committed:
	modified:   app/crud/user_tg.py

Untracked files:
	alembic/versions/2024_11_29_1555-01c576f4b80a_remove_id_field_from_user_group_table.py
	app/admin_endpoints/
	app/templates/

teamofroman reviewed (approved):
Принято

teamofroman committed:
Merge branch 'develop' into fix/crud

teamofroman committed:
Fixed style
